### ISSUE TITLE
[Improvement] Extend expect-expect to support custom functions

### ISSUE THREAD

**Issue by ViktorSoroka07** [veritem/eslint-plugin-vitest#154]
Similar to how it is done in `eslint-plugin-jest`

```
        "jest/expect-expect": [
          "error",
          {
            "assertFunctionNames": ["expect", "test*"]
          }
        ],
```

With that `testSearch` function won't end up in lint error

**Comment by samayer12** [comment]
My team had the same need, so I spent some time to upgrade `expect-expect` using some tutorial repos for eslint rules.

I've got a version of this working [here](https://github.com/samayer12/eslint-typescript-custom-rule/tree/ac4b71f671d36e33838752e184f598bedf567cda) but the reference project I used to learn how to write linter rules is structured differently than this repository.

I'll have to do a little work to port over the implementation and provide additional test cases. I'll spend some time (~2-3hr/wk) on this during a weekly pro-dev block, so expect an update in a couple weeks.
